alicloud/resource_alicloud_cs_managed_kubernetes.go

```diff
@@ -0,0 +1,657 @@
+		})
+		return err
+	}); err != nil {
+		return fmt.Errorf("Waitting for ManagedKubernetes cluster %#v got an error: %#v", cs.Running, err)
+	}
+
+	return resourceAlicloudCSManagedKubernetesUpdate(d, meta)
+}
+
+func resourceAlicloudCSManagedKubernetesUpdate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	d.Partial(true)
+	invoker := NewInvoker()
+	if d.HasChange("worker_numbers") && !d.IsNewResource() {
+
+		workerNumbers := expandIntList(d.Get("worker_numbers").([]interface{}))
+		workerInstanceTypes := expandStringList(d.Get("worker_instance_types").([]interface{}))
+
+		args := &cs.KubernetesClusterResizeArgs{
+			DisableRollback: true,
+			TimeoutMins:     60,
+			LoginPassword:   d.Get("password").(string),
+		}
+
+		if len(workerNumbers) == 1 {
+			args.WorkerInstanceType = workerInstanceTypes[0]
+			args.NumOfNodes = int64(workerNumbers[0])
+		}
+		if err := invoker.Run(func() error {
+			_, err := client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+				return nil, csClient.ResizeKubernetesCluster(d.Id(), args)
+			})
+			return err
+		}); err != nil {
+			return fmt.Errorf("Resize Cluster got an error: %#v", err)
+		}
+
+		if err := invoker.Run(func() error {
+			_, err := client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+				return nil, csClient.WaitForClusterAsyn(d.Id(), cs.Running, 3600)
+			})
+			return err
+		}); err != nil {
+			return fmt.Errorf("Waitting for container Cluster %#v got an error: %#v", cs.Running, err)
+		}
+		d.SetPartial("worker_number")
+	}
+
+	if !d.IsNewResource() && (d.HasChange("name") || d.HasChange("name_prefix")) {
+		var clusterName string
+		if v, ok := d.GetOk("name"); ok {
+			clusterName = v.(string)
+		} else {
+			clusterName = resource.PrefixedUniqueId(d.Get("name_prefix").(string))
+		}
+		if err := invoker.Run(func() error {
+			_, err := client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+				return nil, csClient.ModifyClusterName(d.Id(), clusterName)
+			})
+			if err != nil && !IsExceptedError(err, ErrorClusterNameAlreadyExist) {
+				return err
+			}
+			return nil
+		}); err != nil {
+			return fmt.Errorf("Modify Cluster Name got an error: %#v", err)
+		}
+		d.SetPartial("name")
+		d.SetPartial("name_prefix")
+	}
+	d.Partial(false)
+
+	return resourceAlicloudCSManagedKubernetesRead(d, meta)
+}
+
+func resourceAlicloudCSManagedKubernetesRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	var cluster cs.KubernetesCluster
+	invoker := NewInvoker()
+	if err := invoker.Run(func() error {
+		raw, e := client.WithCsClient(func(csClient *cs.Client) (interface{}, error) {
+			return csClient.DescribeKubernetesCluster(d.Id())
+		})
+		if e != nil {
+			return e
+		}
+		cluster, _ = raw.(cs.KubernetesCluster)
+		return nil
+	}); err != nil {
+		if NotFoundError(err) || IsExceptedError(err, ErrorClusterNotFound) {
```

Comment:
I think `ErrorClusterNotFound ` can be added into NotFoundError(err) by the method DescribeKubernetesCluster.